src/test/java/org/jenkinsci/plugins/gitclient/WorkspaceWithRepoRule.java

```diff
@@ -0,0 +1,144 @@
+package org.jenkinsci.plugins.gitclient;
+
+import hudson.EnvVars;
+import hudson.Launcher;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jgit.lib.ObjectId;
+import org.junit.Rule;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+
+public class WorkspaceWithRepoRule {
+
+    private GitClient gitClient;
+    private File gitFileDir;
+    private CliGitCommand cliGitCommand;
+    protected TaskListener listener;
+    private final String repoURL = "https://github.com/jenkinsci/git-client-plugin.git";
+
+    public WorkspaceWithRepoRule(GitClientSampleRepoRule repo, TaskListener listener) throws Exception {
+        createEnv(repo, null, listener);
+        this.listener = listener;
+    }
+
+    public WorkspaceWithRepoRule(GitClientSampleRepoRule repo, String gitImplName, TaskListener listener) throws Exception {
+        createEnv(repo, gitImplName, listener);
+        this.listener = listener;
+    }
+
+    private void createEnv(GitClientSampleRepoRule rule, String gitImplName, TaskListener listener) throws Exception {
+        gitClient = setupGitClient(rule.getRoot(), gitImplName, listener);
+        gitFileDir = rule.getRoot();
+        cliGitCommand = new CliGitCommand(gitClient);
+    }
+
+    private GitClient setupGitClient(File gitFileDir, String gitImplName, TaskListener listener) throws Exception {
+        if (gitImplName == null){
+            return Git.with(listener, new EnvVars()).in(gitFileDir).getClient();
+        }
+        return Git.with(listener, new EnvVars()).in(gitFileDir).using(gitImplName).getClient();
+    }
+
+    protected String localMirror() throws IOException, InterruptedException {
+        File base = new File(".").getAbsoluteFile();
+        for (File f=base; f!=null; f=f.getParentFile()) {
+            if (new File(f,"target").exists()) {
+                File clone = new File(f, "target/clone.git");
+                if (!clone.exists()) {  // TODO: perhaps some kind of quick timestamp-based up-to-date check?
+                    cliGitCommand.run("git", "clone", "--mirror", repoURL, clone.getAbsolutePath());
```

Comment:
I was actually confused between the use of CliGitCommand or launchCommand (which uses the Launcher to run cmds). 
Since I am using the RepoRule, I thought of using CliGitCommand only to notice that it returns `String[]` instead of a string (which is done by launchCommand). 
To replace launchCommand with gitCommand would have required considerable re-work so I used launchCommand. 

And yes, it was working on my MacBook. Thank you for pointing out the mistake!